### ISSUE TITLE
CKV_AWS_103 - False findings on our AWS ALB

### ISSUE THREAD

**Issue by dbbc96** [bridgecrewio/checkov#5044]
**Describe the issue**
2 days ago our build passes but today got multiple failures.  One of them indicating that to ensure LB is using TLS1.2.  Which it is. 

**Examples**
Here is code snippets from our module which defaults are listed:
```hcl
variable "listener_ssl_policy_default" {
  description = "The security policy if using HTTPS externally on the load balancer. [See](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-security-policy-table.html)."
  type        = string
  default     = "ELBSecurityPolicy-TLS-1-2-Ext-2018-06"
}

resource "aws_lb_listener" "frontend_https" {
  count = var.create_lb ? length(var.https_listeners) : 0

  load_balancer_arn = aws_lb.this[0].arn

  port     = var.https_listeners[count.index]["port"]
  protocol = lookup(var.https_listeners[count.index], "protocol", "HTTPS")
  certificate_arn = var.https_listeners[count.index]["certificate_arn"]
  ssl_policy      = lookup(var.https_listeners[count.index], "ssl_policy", var.listener_ssl_policy_default)
}
```

**Additional context**
The resources hasn't changed and so it had passed 2 days ago and now today it failing on multiple items.  This report is just on the ALB finding.

**Comment by dbbc96** [comment]
to give an update i found that it works fine on 2.3.214 but 220 or 223 it fails on.  223 is the version that our pipeline had installed.

**Comment by gruebel** [comment]
hey @dbbc96 thanks for reaching out.

We recently enabled support for `for_each` in `checkov`, because before we ignored most of the information in your resource definition.

I also need to see the definition for the `var.https_listeners`, because without it we can't investigate the issue.

**Comment by gruebel** [comment]
thanks. I took a look and I think something is off with `lookup` in context of `count`.

**Comment by dbbc96** [comment]
So did you find something wrong on your side?

**Comment by gruebel** [comment]
as I wrote, currently the combination of `count` + `lookup` doesn't work correctly.

**Comment by dbbc96** [comment]
so what will be needed to resolve this issue?
